chunkator/__init__.py

```diff
@@ -11,9 +11,12 @@ def chunkator(queryset, chunk_size):
     length. We're saving at least a ``count()`` query on QuerySets, or a
     CPU-and-RAM-consuming ``len(queryset)`` query.
     """
-    pk = 0
+    pk = None
     while True:
-        page = queryset.filter(pk__gt=pk).order_by('pk')[:chunk_size]
+        queryset = queryset.order_by('pk')
```

Comment:
And this is a wron way to do, in most case you do not need to order your result (export, batch, ...) and in these case you **MUST** order on the pk, as is you can do key pagination, this is what we need here, for memory you can re-read http://tech.novapost.fr/django-paginator-offset-keyset-pagination.html